server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -260,6 +260,17 @@ public Iterator<Setting<?>> settings() {
         Property.IndexScope
     );
 
+    public static final String SETTING_SEGMENT_REPLICATION = "index.replication.segment_replication";
```

Comment:
Nitpick, but I'd consider making this an enum setting, like `index.replication.type=["DOCUMENT"|"SEGMENT"]`, defaulting to "DOCUMENT". It conveys a little more information that the replication methods are mutually exclusive for a given index, and also gives a name to "document replication" instead of being implicit.